Lib/Src/CJsonParser.cpp

```diff
@@ -0,0 +1,339 @@
+                {
+                    current_token.text.append(1, current_char);
+                }
+                else
+                {
+                    endToken(current_token);
+                    current_token.text.append(1, current_char);
+                    current_token.type = INTEGER;
+                }
+                break;
+            case '.':
+                if (current_token.type == INTEGER)
+                {
+                    current_token.text.append(1, current_char);
+                    current_token.type = FLOAT;
+                }
+                else
+                {
+                    endToken(current_token);
+                    current_token.text.append(1, current_char);
+                    current_token.type = INVALID;
+                }
+                break;
+            case ' ':
+            case '\t':
+                if (current_token.type == STRING)
+                {
+                    current_token.text.append(1, current_char);
+                }
+                else
+                {
+                    endToken(current_token);
+                }
+                break;
+            case '\n':
+            case '\0':
+            case '\r':
+                endToken(current_token);
+                break;
+            case ':':
+                addSpecialChar(current_token, COLON, current_char);
+                break;
+            case '{':
+                addSpecialChar(current_token, CURLY_OPEN, current_char);
+                break;
+            case '}':
+                addSpecialChar(current_token, CURLY_CLOSE, current_char);
+                break;
+            case '[':
+                addSpecialChar(current_token, ARRAY_OPEN, current_char);
+                break;
+            case ']':
+                addSpecialChar(current_token, ARRAY_CLOSE, current_char);
+                break;
+            case ',':
+                addSpecialChar(current_token, COMMA, current_char);
+                break;
+
+            case '"':
+                if (current_token.type != STRING)
+                {
+                    endToken(current_token);
+                    current_token.type = STRING;
+                }
+                else
+                {
+                    endToken(current_token);
+                }
+                break;
+
+            default:
+                if (current_token.type == WHITESPACE ||
+                    current_token.type == INTEGER ||
+                    current_token.type == FLOAT)
+                {
+                    endToken(current_token);
+                    current_token.text.append(1, current_char);
+                    current_token.type = INVALID;
+                }
+                else
+                {
+                    current_token.text.append(1, current_char);
+                }
+                break;
+        }
+    }
+    // Finish any token that might have not been stored
```

Comment:
haha. you even put finish here in the comment, yet the name of the method does not match it's function.